bigquery/src/main/scala/com/yahoo/maha/executor/bigquery/BigqueryQueryExecutor.scala

```diff
@@ -0,0 +1,174 @@
+import com.github.vertical_blank.sqlformatter.scala.{FormatConfig, SqlDialect, SqlFormatter}
+import com.google.auth.oauth2.ServiceAccountCredentials
+import com.google.cloud.ServiceOptions
+import com.google.cloud.bigquery.BigQuery
+import com.google.cloud.bigquery.BigQueryOptions
+import com.google.cloud.bigquery.JobId
+import com.google.cloud.bigquery.JobInfo
+import com.google.cloud.bigquery.QueryJobConfiguration
+import com.yahoo.maha.core._
+import com.yahoo.maha.core.query._
+import grizzled.slf4j.Logging
+import java.io.FileInputStream
+import java.net.{Authenticator, PasswordAuthentication}
+import java.util.UUID
+import scala.util.Try
+
+private case class ProxyCredentials(
+  @JsonProperty("proxy_user") userName: String,
+  @JsonProperty("proxy_pass") password: String
+)
+
+case class BigqueryQueryExecutorConfig(
+  gcpCredentialsFilePath: String,
+  gcpProjectId: String,
+  enableProxy: Boolean,
+  proxyCredentialsFilePath: Option[String],
+  proxyHost: Option[String],
+  proxyPort: Option[String],
+  retries: Int
+) extends Logging {
+
+  private final val HttpsProxyHost = "https.proxyHost"
+  private final val HttpsProxyPort = "https.proxyPort"
+  private final val DisabledAuthSchemes = "jdk.http.auth.tunneling.disabledSchemes"
+
+  private val mapper = {
+    val om = new ObjectMapper(new YAMLFactory()) with ScalaObjectMapper
+    om.registerModule(DefaultScalaModule)
+    om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    om
+  }
+
+  def buildBigqueryClient(): BigQuery = {
+    BigQueryOptions
+      .newBuilder()
+      .setCredentials(ServiceAccountCredentials.fromStream(new FileInputStream(gcpCredentialsFilePath)))
+      .setProjectId(gcpProjectId)
+      .setRetrySettings(ServiceOptions.getDefaultRetrySettings().toBuilder().setMaxAttempts(retries).build())
+      .build()
+      .getService
+  }
+
+  def configureProxy(): Unit = {
+    if (enableProxy && proxyCredentialsFilePath.isDefined && proxyHost.isDefined && proxyHost.isDefined) {
+      val proxyCredentials = mapper.readValue[ProxyCredentials](
+        new FileInputStream(proxyCredentialsFilePath.get)
+      )
+
+      System.setProperty(HttpsProxyHost, proxyHost.get)
+      System.setProperty(HttpsProxyPort, proxyPort.get)
+      System.setProperty(DisabledAuthSchemes, "")
+
+      Authenticator.setDefault(new Authenticator() {
+        override protected def getPasswordAuthentication =
+          new PasswordAuthentication(proxyCredentials.userName, proxyCredentials.password.toCharArray)
+      })
+    } else {
+      info("Not using proxy for BigQuery Query Execution")
+    }
+  }
+}
+
+class BigqueryQueryExecutor(
+  config: BigqueryQueryExecutorConfig,
+  lifecycleListener: ExecutionLifecycleListener
+) extends QueryExecutor
+  with Logging {
+
+  val engine: Engine = BigqueryEngine
+  config.configureProxy()
+  lazy val bigqueryClient = config.buildBigqueryClient()
```

Comment:
Is this client builder meant to be lazy?  may not fail until first access but after Executor instance created successfully.

Reply:
Had it as `lazy` initially so that we could instantiate the Executor in the ExecutoryFactory tests without needing to pass valid credentials files, but I've removed it now.